File: app/src/main/java/com/robotnec/budget/navigator/AndroidNavigationBundle.java
package com.robotnec.budget.navigator;

import android.support.v7.app.AppCompatActivity;

import java.io.Serializable;

import com.robotnec.budget.core.navigator.NavigationBundle;

/**
 * @author zak <zak@swingpulse.com>
 */
public class AndroidNavigationBundle implements NavigationBundle<AppCompatActivity> {

    private AppCompatActivity activity;
    private Serializable serializableExtra;

    public AndroidNavigationBundle(AppCompatActivity activity) {
        this.activity = activity;
    }

    @Override
    public AppCompatActivity getNavigationContext() {
        return activity;
    }

    @Override
    public Serializable getSerializableExtra() {
        return serializableExtra;
    }

    @Override
    public void setSerializableExtra(Serializable serializableExtra) {
        this.serializableExtra = serializableExtra;
    }
}

File: app/src/main/java/com/robotnec/budget/app/navigator/AndroidNavigationBundle.kt
package com.robotnec.budget.app.navigator

import androidx.appcompat.app.AppCompatActivity
import com.robotnec.budget.core.navigator.NavigationBundle
import java.io.Serializable

/**
 * @author zak zak@swingpulse.com>
 */
@Deprecated("TODO rewrite to command-visitor")
class AndroidNavigationBundle(override val navigationContext: AppCompatActivity,
                              override var serializableExtra: Serializable?) : NavigationBundle<AppCompatActivity>
